mesonbuild/backend/backends.py

```diff
@@ -323,26 +319,59 @@ def _flatten_object_list(self, target, objects, proj_dir_to_build_root):
                 raise MesonException('Unknown data type in object list.')
         return obj_list
 
-    def serialize_executable(self, tname, exe, cmd_args, workdir, env=None,
-                             extra_paths=None, capture=None):
+    def as_meson_exe_cmdline(self, tname, exe, cmd_args, for_machine=MachineChoice.BUILD,
+                             extra_bdeps=[], capture=None):
```

Comment:
__[Flake8]__ [B006] Do not use mutable data structures for argument defaults. All calls reuse one instance of that data structure, persisting changes between them. ([link](https://sider.review/gh/repos/19784232/pulls/5644#issue-4339557))

<sub>posted by Sider</sub>